packages/core/src/domain/patient.ts

```diff
@@ -23,20 +23,13 @@ export type Period =
     };
 
 export type BaseIdentifier = {
+  value: string;
   period?: Period;
   assigner?: string;
 };
-// TODO #425 reenable this when we manage to work with diff systems @ CW
-// export type PersonalIdentifier = BaseIdentifier &
-//   (
-//     | { type: GeneralTypes; value: string; state?: never }
-//     | { type: DriverLicenseType; value: string; state: USState }
-//   );
-export type PersonalIdentifier = BaseIdentifier & {
-  type: DriverLicenseType;
-  value: string;
-  state: USState;
-};
+
+export type PersonalIdentifier = BaseIdentifier &
+  ({ type: GeneralTypes } | { type: DriverLicenseType; state: USState });
```

Comment:
I don't remember why we had the type structured that way on that comment, so I'm wondering why don't we just have a single type with all types of identifiers there (driversLicense, passport, ssn, medicare)?

Probably so we can have the conversion maps w/ strong types for drivers licence per state?

Reply:
Yes, I think this mimics that zod `or` functionality and with strong types for the state field in the case of drivers license. Happy to implement a different suggestion.